File: JAVA PROGRAMMING/Test48.java
// 48. nCr & nPr IN JAVA
import java.util.Scanner;
public class Test {
    public static int factorial(int n) {
        int temp = 1;
        for(int i = 1; i <= n; i++) {
            temp *= i;
        }
        return temp;
    }
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        System.out.println("Enter N Value: ");
        int n = in.nextInt();

        System.out.println("Enter R Value: ");
        int r = in.nextInt();

        int factn = factorial(n);
        int factr = factorial(r);
        int factnr = factorial(n-r);

        int nCr = (factn / (factr * factnr));
        int nPr = (factn / factnr);

        System.out.println(n+"C"+r+" Is: "+nCr);
        System.out.println(n+"P"+r+" Is: "+nPr);

        in.close();
    }
}

File: TYPESCRIPT PROGRAMMING/Test48.ts
// 48. nCr & nPr IN JAVA
const fact = (ip: number): number => {
    let temp: number = 1;
    for(let i: number = 1; i <= ip; i++) {
        temp *= i;
    }
    return temp;
}
let n: number = 5;
let r: number = 3;
let nr: number = n - r;
let nfact = fact(n);
let rfact = fact(r);
let nrfact = fact(nr);
let nCr = nfact / (rfact * (nrfact));
let nPr = nfact / nrfact;
console.log(`${n}C${r} : ${nCr}`);
console.log(`${n}P${r} : ${nPr}`);